### PR TITLE
Add dependabot/codeql/mergify

### PR DESCRIPTION
**Pull request by jpkrohling**
This PR adds the capability of automatic merge of npm package updates, as long as they pass the regular tets.

Signed-off-by: Juraci Paixão Kröhling <juraci@kroehling.de>

### CODE REVIEW

File: .github/mergify.yml
Comment (yurishkuro) [inline .github/mergify.yml:8]: Listing all merge conditions like this is kind of unfortunate. If we add another required check in the future, we'd need to remember to add it here too.

Does it not support "merge on green" where "green" is defined by the required checks on the repo, whatever they are?

**Review by yurishkuro — APPROVED** [review]

File: .github/mergify.yml
Comment (jpkrohling) [reply .github/mergify.yml:8]: It might now, I think we could determine the conditions on the repo settings and mergify just won't be able to merge it. We could give it a try afterward.

Comment (jpkrohling) [reply .github/mergify.yml:8]: >Note that Mergify will always respect the branch protection settings. When the conditions match and the merge action runs, Mergify waits for the branch protection to be validated before merging the pull request.

It might just work, I'll remove this from this PR.

File: .github/mergify.yml
Comment (yurishkuro) [inline .github/mergify.yml:4]: I assume this will require manual approval before auto-merging.

We need to be more careful with this repo than with compiled languages, as we do not have full code coverage, and UI features sometimes have subtle functional differences.

Comment (jpkrohling) [reply .github/mergify.yml:4]: Yes. I removed this requirement, but just reinstated based on your comment.

image

**Pull request merged**
